src/linker/Linker/Message.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Mono.Linker
+{
+	public class Message
+	{
+		public MessageOrigin Origin { get; }
+		
+		public string Subcategory { get; }
+		
+		public MessageCategory Category { get; }
+		
+		public MessageCode Code { get; }
```

Comment:
What is this for?

Reply:
MSBuild requires us to identify all warnings and errors with a code (e.g. CS0029).